rusty_pizza_server/src/order_model/meal.rs

```diff
@@ -119,29 +130,34 @@ mod tests {
                 variety: String::from("groß"),
                 price: Money::new(5, 50),
                 specials: expected_specials,
+                special_factory: expected_special_factory,
             }
         );
     }
 
     #[test]
     fn special_can_be_removed_from_meal() {
         //Given
+        let mut special_factory = SpecialFactory::new();
         let mut specials = BTreeSet::new();
-        specials.insert(String::from("Käserand"));
+        specials.insert(special_factory.create_special(String::from("Käserand")));
         let mut meal = Meal {
             id: 0,
             meal_id: String::from("03"),
             variety: String::from("groß"),
             price: Money::new(5, 50),
             specials,
+            special_factory,
         };
 
-        let special = String::from("Käserand");
+        let special = Special::new(0, String::from("Käserand"));
```

Comment:
Is it safe that the special created before always gets the ´ID´ 0?

I would do somethin like:
```suggestion
        let special = special_factory.create_special(String::from("Käserand"))
        [...]
        specials.insert(special);
        [...]
        meal.remove_special(&special);
```
Use the same special for insert and remove. We do not need to test the behavior of specials here

Reply:
This test case was removed because we only remove by ID and not by reference.